libs/clients/university-of-iceland/src/lib/universityOfIcelandClient.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import { Injectable } from '@nestjs/common'
+import {
+  StudentTranscriptApi,
+  NemandiGetLocaleEnum,
+  Transcripts,
+  NemandiFerillFerillGetLocaleEnum,
+  StudentTrackOverview,
+  NemandiFerillFerillFileTranscriptGetLocaleEnum,
+} from '../../gen/fetch'
+
+@Injectable()
+export class UniversityOfIcelandService {
+  constructor(private readonly universityOfIcelandApi: StudentTranscriptApi) {}
+
+  private universityOfIcelandApiWithAuth = (user: User) =>
+    this.universityOfIcelandApi.withMiddleware(new AuthMiddleware(user as Auth))
+
+  async studentInfo(
+    user: User,
+    locale?: NemandiGetLocaleEnum,
+  ): Promise<Transcripts> {
+    return await this.universityOfIcelandApiWithAuth(user).nemandiGet({
+      locale: locale,
+    })
+  }
```

Comment:
FYI the `await` is redundant the promise can just be returned.

```suggestion
  studentInfo(
    user: User,
    locale?: NemandiGetLocaleEnum,
  ): Promise<Transcripts> {
    return this.universityOfIcelandApiWithAuth(user).nemandiGet({
      locale: locale,
    })
  }
```